src/Shared/Buffers.MemoryPool/SlabMemoryPool.cs

```diff
@@ -160,6 +160,18 @@ internal void Return(MemoryPoolBlock block)
             }
         }
 
+        internal void RefreshBlock(MemoryPoolSlab slab, int offset, int length)
+        {
+            lock (_disposeSync)
+            {
+                if (!_isDisposed && slab != null && slab.IsActive)
+                {
+                    // Need to make a new object because this one is being finalized
```

Comment:
Maybe add a comment indicating that the MemoryPoolBlock ctor itself must be called with the _disposeSync lock.